test_back_copper_only_routing test has been failing since added

## Summary

The test `tests/test_router_core.py::TestSingleLayerRouting::test_back_copper_only_routing` has been failing since it was originally added in commit a3cc708 ("feat(router): Add allowed_layers constraint for single-layer routing").

## Test Failure Details

```
AssertionError: Segment on Layer.F_CU, expected B.Cu
```

The test creates through-hole pads with `allowed_layers=["B.Cu"]` but the router generates segments on F.Cu instead of B.Cu.

## Root Cause

When routing between PTH pads with layer constraints, the router is not properly filtering the starting layer. Even though `allowed_layers=["B.Cu"]` is set, the route starts on F.Cu.

## Expected Behavior

When `allowed_layers=["B.Cu"]` is configured, all route segments should be on B.Cu, even for through-hole pads that could theoretically be accessed from any layer.

## Discovered During

Code review of PR #958 (feat(router): Handle off-grid pads by expanding goal region).

This is NOT a regression from that PR - the test was already failing before those changes were made.

--- a/src/kicad_tools/router/pathfinder.py
+++ b/src/kicad_tools/router/pathfinder.py
@@ -461,8 +461,10 @@ class Router:
         if self.rules.allowed_layers is None:
             return True  # No restriction
 
-        layer_name = self.grid.index_to_layer(layer_idx)
-        return layer_name in self.rules.allowed_layers
+        # Convert grid index to Layer enum value, then to KiCad name for comparison
+        layer_value = self.grid.index_to_layer(layer_idx)
+        layer = Layer(layer_value)
+        return layer.kicad_name in self.rules.allowed_layers
 
     def _can_place_via_in_zones(self, gx: int, gy: int, net: int) -> bool:
         """Check if via placement is legal considering zones on all layers.

